worlds/factorio/Client.py

```diff
@@ -521,10 +521,14 @@ def _handle_color(self, node: JSONMessagePart):
 rcon_password = args.rcon_password if args.rcon_password else ''.join(
     random.choice(string.ascii_letters) for x in range(32))
 factorio_server_logger = logging.getLogger("FactorioServer")
-options = Utils.get_settings()
-executable = options["factorio_options"]["executable"]
+settings: FactorioSettings = get_settings().factorio_options
+if os.path.samefile(settings.executable, sys.executable):
+    raise Exception(f"FactorioClient was set to run itself {settings.executable}, aborting process bomb.")
```

Comment:
I'd suggest to clear the settings value so the file selector reappears the next time
```suggestion
    selected_executable = settings.executable
    settings.executable = FactorioSettings.executable  # reset to default
    raise Exception(f"FactorioClient was set to run itself {selected_executable}, aborting process bomb.")
```